Cppcheck flagging BUGCHECK macro as syntax error

**Describe the bug**
The use of the `BUGCHECK` macro with the second argument blank looks suspicious and it is reported as a syntax error by newer versions of the `cppcheck` tool:

```
Checking src/os/shared/src/osapi-printf.c ...
src/os/shared/src/osapi-printf.c:269:30: error: syntax error [syntaxError]
    BUGCHECK((String) != NULL, )
                             ^
```

**To Reproduce**
Execute a recent version of `cppcheck` against OSAL source code - observed in version 2.7.

**Expected behavior**
Should pass cleanly

**System observed on:**
Ubuntu 22.04

**Additional context**
The second argument being blank here is intentional, because it is a return value and this is a void function, so it needs to be empty.  The syntax is actually valid but it is just a bit unusual.

**Reporter Info**
Joseph Hickey, Vantage Systems, Inc.

--- a/src/os/inc/osapi-macros.h
+++ b/src/os/inc/osapi-macros.h
@@ -136,4 +136,13 @@
  */
 #define LENGTHCHECK(str, len, errcode) ARGCHECK(memchr(str, '\0', len), errcode)
 
+/**
+ * @brief Bug-Check macro for void functions
+ *
+ * The basic BUGCHECK macro returns a value, which needs to be empty
+ * for functions that do not have a return value.  In this case the
+ * second argument (errcode) is intentionally left blank.
+ */
+#define BUGCHECK_VOID(cond) BUGCHECK(cond, )
+
 #endif /* OSAPI_MACROS_H */

--- a/src/os/shared/src/osapi-printf.c
+++ b/src/os/shared/src/osapi-printf.c
@@ -266,7 +266,7 @@ void OS_printf(const char *String, ...)
     char    msg_buffer[OS_BUFFER_SIZE];
     int     actualsz;
 
-    BUGCHECK((String) != NULL, )
+    BUGCHECK_VOID(String != NULL)
 
     if (OS_SharedGlobalVars.GlobalState != OS_INIT_MAGIC_NUMBER)
     {

